src/pytorch_lightning/trainer/trainer.py

```diff
@@ -696,6 +696,8 @@ def fit(
 
             datamodule: An instance of :class:`~pytorch_lightning.core.datamodule.LightningDataModule`.
         """
+        if not isinstance(model, pl.LightningModule):
```

Comment:
do you think we can keep this inside `_run` method? just to avoid duplicate code. There we can access the method call type using `self.state.fn`

Reply:
No, we want to fail early before creating processes.